src/Juvix/Compiler/Core/Data/InfoTable.hs

```diff
@@ -12,7 +12,9 @@ data InfoTable = InfoTable
     _infoIdentifiers :: HashMap Symbol IdentifierInfo,
     _infoInductives :: HashMap Symbol InductiveInfo,
     _infoConstructors :: HashMap Tag ConstructorInfo,
-    _infoAxioms :: HashMap Name AxiomInfo
+    _infoAxioms :: HashMap Name AxiomInfo,
+    _infoSymbolsNum :: Word,
```

Comment:
I would prefer the name `infoNextSymbol` instead or something similar. It clarifies that this field is only used for fresh names and does never need to be decreased.
Same for tags.

Reply:
Done